spyder/plugins/remoteclient/widgets/connectiondialog.py

```diff
@@ -426,28 +424,127 @@ def _create_configfile_subpage(self):
         configfile = self.create_browsefile(
             text=_("Configuration file *"),
             option=f"{self.host_id}/configfile",
+            tip=_("File with the OpenSSH client configuration to use"),
             alignment=Qt.Vertical,
             status_icon=ima.icon("error"),
         )
 
+        host = self.create_lineedit(
+            text=_("Host *"),
+            option=f"{self.host_id}/{AuthenticationMethod.ConfigFile}/address",
+            default="",
+            tip=_(
+                "This is the host (<i><b>Host</b></i> configuration keyword) "
+                "that encapsulates the options to use for the connection"
+            ),
+            status_icon=ima.icon("error"),
+        )
+
+        port = self.create_spinbox(
+            prefix=_("Port"),
+            suffix="",
+            option=f"{self.host_id}/{AuthenticationMethod.ConfigFile}/port",
+            min_=0,
+            max_=65535,
+            tip=_("Introduce a port to use for this connection. "
+                  "Set <i><b>0</b></i> (<i><b>From file</b></i> value) to use "
+                  "the port specified in the configuration file"),
+        )
+        port.spinbox.setSpecialValueText(_("From file"))
+        port.spinbox.setStyleSheet("margin-left: 5px")
+
+        username = self.create_lineedit(
+            text=_("Username"),
+            option=f"{self.host_id}/{AuthenticationMethod.ConfigFile}/username",
+            status_icon=ima.icon("error"),
+        )
+
+        password = self.create_lineedit(
+            text=_("Password"),
+            option=f"{self.host_id}/password",
+            tip=(
+                _("Your password will be saved securely by Spyder")
+                if self.NEW_CONNECTION
+                else _("Your password is saved securely by Spyder")
+            ),
+            status_icon=ima.icon("error"),
+            password=True
+        )
+
+        keyfile = self.create_browsefile(
+            text=_("Key file"),
+            option=f"{self.host_id}/keyfile",
+            alignment=Qt.Vertical,
+            status_icon=ima.icon("error"),
+        )
+
+        passphrase = self.create_lineedit(
+            text=_("Passphrase"),
+            option=f"{self.host_id}/passphrase",
+            tip=(
+                _("Your passphrase will be saved securely by Spyder")
+                if self.NEW_CONNECTION
+                else _("Your passphrase is saved securely by Spyder")
+            ),
+            password=True
+        )
+
         validation_label = MessageLabel(self)
 
         # Add widgets to their required dicts
         self._name_widgets[AuthenticationMethod.ConfigFile] = name
         self._widgets_for_validation[AuthenticationMethod.ConfigFile] = [
             name,
+            host,
```

Comment:
```suggestion
            host,
            username,
            password,
            keyfile,
            passphrase,
```

Add these other widgets to perform more validations:

* If a user introduces a `username`, then we need to check that `password` is not empty, and viceversa.
* If they introduce `passphrase`, then `keyfile` can't be empty, and viceversa.
* They can't introduce `username/password` and `keyfile/passphrase` at the same time because only one method is necessary for authentication.
* We should check that none of those fields are available in the SSH config file (if possible). Otherwise, we should show a warning about it.

Please implement those validations in the `validate_page` method.

Reply:
So thinking about this validations, I think there are a couple of cases that should work and kind of go against the validations listed:

* Providing only password when the config file provides the username
* Not providing a passphrase when the config file provides the keyfile or even when the keyfile is provided by the user inside the connections dialog but the keyfile doesn't require a passphrase.

Is there a reason to not allow such cases? Or maybe I'm not fully understanding the extra validations definitions? Also, following the config file validation item, the idea is that users should not set the info that can be provided via the connection dialog in the config file (i.e values for username and keyfile)?

Reply:
> So thinking about these validations, I think there are a couple of cases that should work and kind of go against the validations listed:
>
> * Providing only password when the config file provides the username
> * Not providing a passphrase when the config file provides the keyfile or even when the keyfile is provided by the user inside the connections dialog but the keyfile doesn't require a passphrase.
>
> Is there a reason to not allow such cases?

Nop, I just didn't think about them. The validations I proposed should work when username/keyfile are not present in the config file.

> Also, following the config file validation item, the idea is that users should not set the info that can be provided via the connection dialog in the config file (i.e values for username and keyfile)? Or maybe the idea is to somehow add a message when this happens (values where provided via the dialog and also a value is available from the config file)?

I think we need to parse the config file and fill those fields (i.e. username and keyfile) if they're present in it, and make them read-only (see below). Also, when the passphrase is not required in the config file, then we should disable that field too.

> From my understanding the config file values act as a fallback to the other values so when a value is not provided (in our case no value is given via de connection dialog fields) asyncssh tries to get the missing options from the config file provided.

I understand that, but I think config files are distributed to people to make complex SSH connections. So, in that case we should load the info present in them to our UI and not allow users to edit it so they can perform a successful connection. If they want to change that info, then I'd say they should edit the config file (or request a new one from IT).

Reply:
What do you think above the above way to make things work @ccordoba12 ?

Reply:
I think so, from what I remember and checking the code here, I think I used `SSHClientConfig.load` to parse the selected config file and validate if it was possible to parse it. Also, if the file was parsed successfully, then use any parsed values available to set placeholders for the page widgets